src/java/org/apache/cassandra/db/compaction/unified/Controller.java

```diff
@@ -432,16 +432,16 @@ public int readAmplification(long length, int scalingParameter)
 
         int F = scalingParameter < 0 ? 2 - scalingParameter : 2 + scalingParameter;
         int T = scalingParameter < 0 ? 2 : F;
-        int maxIndex = Math.max(0, (int) Math.floor((Math.log(length) - Math.log(m)) / (Math.log(F) - Math.log(o))));
+        int maxIndex = maxBucketIndex(length, F);
 
         int ret = 0;
-        for (int i = 0; i < maxIndex; i++)
+        for (int i = 0; i <= maxIndex; i++)
```

Comment:
Aren't we now adding the last level twice? Once here and once with the special treatment below?